src/coding-guidelines/expressions.rst

```diff
@@ -82,6 +82,69 @@ Expressions
 
          fn with_base(_: &Base) { ... }
 
+.. guideline:: Do not use integer type as divisor
+   :id: gui_7y0GAMmtMhch
+   :category: advisory
+   :status: draft
+   :release: latest
+   :fls: fls_Q9dhNiICGIfr
+   :decidability: decidable
+   :scope: module
+   :tags: numerics, subset
+
+   This guideline applies when a `Division Expression
+   <https://rust-lang.github.io/fls/expressions.html#syntax_divisionexpression>`_ or `RemainderExpression
+   <https://rust-lang.github.io/fls/expressions.html#syntax_remainderexpression>`_ is used with a RightOperand of
+   `integer type <https://rust-lang.github.io/fls/types-and-traits.html#integer-types>`_.
+
+   .. rationale::
+      :id: rat_vLFlPWSCHRje
+      :status: draft
+
+      The built-in semantics for these expressions can result in panics when division by zero occurs. It is
+      recommended to either:
+
+      * Use checked division functions, which ensure the programmer handles the case when the divisor is zero, or
+      * To create divisors using :std:`std::num::NonZero`, which then allows the programmer to perform those
+        operations knowing that their divisor is not zero.
+
+      Since the compiler can assume the value of :std:`std::num::NonZero` is not zero, the check for zero
+      while dividing with standard integer types can be elided.
+
+   .. non_compliant_example::
+      :id: non_compl_ex_0XeioBrgfh5z
+      :status: draft
+
+      When either the division or remainder are performed, the right operand is evaluated to zero and the
+      program panics.
+
+      .. code-block:: rust
+
+         let x = 0;
+         let y = 5 / x;
+         let x = 5 % x;
```

Comment:
```suggestion
         let z = 5 % x; // This line would also panic.
```

Added the same kind of comment as above, plus changed the variable to `z` to keep things extra clear (by not using `x` anywhere but as a divisor)